tgfloofbot/plugins/administration/models.py

```diff
@@ -0,0 +1,18 @@
+from sqlalchemy import Column, Integer, String, text, DateTime, Boolean
+import datetime
+
+from ...models import ORMBase
+
+
+class Warning(ORMBase):
+    __tablename__ = "tg_warnings"
+    id = Column(Integer, primary_key=True, nullable=False)
+    user_id = Column(Integer, nullable=False)
+    date_added = Column(DateTime, default=datetime.datetime.utcnow, nullable=False)
```

Comment:
So this is a bit of a tricky one. I don't think SQLAlchemy + Alembic is smart enough to properly create the defaults for a `utcnow` call when generating migrations. It's also not something I suggest using for getting a timezone aware UTC datetime object as I've [mentioned before](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow), so I suggest actually just removing the default here.

Reply:
is it better to remove default and allow nullable or change default to datetime,datetime,now(timezone,utc)?

Reply:
Hm... I don't think it should be nullable. Additionally, the `default` field would have to be a function, not a value. It's a bit annoying, but we'd have to do something like this:

```py
def tz_aware_now() -> datetime.datetime:
    return datetime.datetime.now(datetime.timezone.utc)


class Warning(ORMBase):
    ...
    date_added = Column(DateTime, default=tz_aware_now, nullable=False)
    ...
```